File: false_position.py
def fun(x):
    return x ** 3 - x ** 2 + 2

def regulafalsemethod(a, b):
    if fun(a) * fun(b) >= 0:
        print("Root doesn't exist in this interval.")
        return

    c, prev_c = 0, 0
    error = 0
    iterator = 1

    print(f"I\ta\tb\tc\terror")

    while True:
        c = (a * fun(b) - b * fun(a)) / (fun(b) - fun(a))
        error = abs((c - prev_c) / c) * 100
        print(f"{iterator}\t{a:.2f}\t{b:.2f}\t{c:.2f}\t{error:.2f}")

        if fun(c) == 0:
            break
        elif fun(c) * fun(a) > 0:
            a = c
        else:
            b = c

        prev_c = c
        iterator += 1

        if error < 0.01:
            break

    print(f"The value of root is: {c:.2f}")



a, b = -5,12
regulafalsemethod(a, b)

File: false_position.java
public class FalsePosition {
    public static double f(double x) {
        return 3 * x - Math.cos(x) - 1;
    }
    public static void falsePosition(double a, double b, double tolerance) {
        if (f(a) * f(b) >= 0) {
            System.out.println("Root does not exist.");
            return;
        }
        double c = 0, error = 1;
        while (error > tolerance) {
            c = (a * f(b) - b * f(a)) / (f(b) - f(a));
            error = Math.abs(f(c));
            if (f(c) == 0) {
                break;
            }
            if (f(a) * f(c) > 0) {
                a = c;
            } else {
                b = c;
            }
        }
        System.out.println("The root is: " + c);
    }
    public static void main(String[] args) {
        falsePosition(0, 1, 0.001);
    }
}
